lib/ansible/module_utils/k8s/raw.py

```diff
@@ -91,24 +99,29 @@ def execute_module(self):
             }
         })
 
-    def perform_action(self, resource, definition):
+    def perform_action(self, resource, definition, facts=False):
         result = {'changed': False, 'result': {}}
         state = self.params.pop('state', None)
         force = self.params.pop('force', False)
         name = definition.get('metadata', {}).get('name')
         namespace = definition.get('metadata', {}).get('namespace')
+        labels = ','.join(['%s=%s' % (k, v) for (k, v) in definition.get('metadata', {}).get('labels', {}).items()])
         existing = None
 
         self.remove_aliases()
 
-        if definition['kind'].endswith('list'):
-            result['result'] = resource.get(namespace=namespace).to_dict()
+        if facts:
+            response = resource.get(namespace=namespace, name=name, label_selector=labels).to_dict()
+            if 'items' in response:
+                result['result'] = response
+            else:
+                result['result'] = dict(items=[response])
             result['changed'] = False
             result['method'] = 'get'
             return result
 
         try:
-            existing = resource.get(name=name, namespace=namespace)
+            existing = resource.get(name=name, namespace=namespace, label_selector=labels)
```

Comment:
If we want to support selectors we should probably just let the user pass them in, and the client supports fieldSelectors as well as labelSelectors